src/transformStateWithClones.js

```diff
@@ -1,13 +1,48 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object} stateClone
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateClone = {};
+
+  Object.assign(stateClone, state);
+
+  const resultArray = [];
+
+  for (const action of actions) {
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        resultArray.push(Object.assign({}, stateClone));
+        break;
+
+      case 'removeProperties':
+        for (const property in stateClone) {
+          if (action.keysToRemove.includes(property)) {
+            delete stateClone[property];
+          }
```

Comment:
you don't need `if` here, if key doesn't exist in the object then delete will ignore this key